IPs/RSKIP178.md

```diff
@@ -0,0 +1,375 @@
+		return (false,false)
+	else
+		return (false,true)
+else	
+	expected = "RSKBLOCK:"+ RSKHashForMergedMining
+	rskTagPosition = last Index Of expected in tail
+	if (rskTagPosition == -1)  return (true,false)
+	if (rskTagPosition >= 64)  return (true,false)
+	if (rskTagPosition !=lastTag) return (true,false)
+	if (tail.length - rskTagPosition > 169) return (true,false)
+
+coinbaseLength = tail.length + byteCount;
+if (coinbaseLength <= 64) return (mergeMined,false)
+
+transactionHash = SHA256(tail) starting from the midstate40 
+coinbaseTransactionHash = reverseBytes(transactionHash)
+
+if (not validMerkleTree(merkleRoot,coinbaseTransactionHash)) return (mergeMined,false)
+return (mergeMined,true)
+```
+
+This logic of the *getMMInfo()* function is very similar to the existent in the *ProofOfWorkRule* in rskj and the *ProofOfWorkRule* must be changed to use this function.
+
+If (isMMValid==true) and (mergeMined==true), then the input represents an **anchor heavyblock** (it can be an also an internal uncle, and external uncle, or a mainchain block). 
+
+If (isMMValid==true) and (mergeMined==false), then the input represents an **confirmation heavyblock**. 
+
+Anchor heavyblocks can be referenced by three distinct means:
+
+- in normal mainchain RSK block, using the mainchain block merge-mining proof
+- in an RSK uncle block header, using the uncle's merge-mining proof, representing a **internal uncle anchor heavyblock**. 
+- in the *heavyblocks* list, also representing an **external uncle anchor heavyblock**, but one that has been  extracted from the Bitcoin blockchain, and not from the RSK network.
+
+To identify **external uncle anchor heavyblocks** from the Bitcoin blockchain, the ARMADILLO data must be extended to include at least 14 bytes of the RSK parent block id, or 4 bytes of the Bitcoin block id associated with it. Until ARMADILLO is improved, this RSKIP proposes that the ARMADILLO tag as it is today is used to identify the uncle. This will extend the uncle reach to a depth of 64 at most, which facilitates "nothing-at-stake" double-spends for up to depth 64. By including just two bytes of the Bitcoin pow of the prior RSK block in the ARMADILLO data, the attack probability is reduced 2^16-fold. In other words, considering 50% of the hashrate merge-mining RSK, the attack opportunity will only occur spontaneusly once every 2.5 years.
+
+Note that the midstate of *SHA256Digest* class in rskj uses a different format that the one used in this description. We call our variable midstate40 to distinguish between them.
+
+Also it is important to note that the above algorithm requires that for a block that doesn't have the merge-mining the tail size must be equal or grater than 169 bytes, but less than 233 bytes or the midstate is in fact the SHA256 initial state, while for a header that does have the merge mining tag, the tail size must be less than 169 bytes. This last check is the same as how the current check *ProofOfWorkRule* performs as:
```

Comment:
Also it is important to note that the above algorithm requires that for a block that doesn't have the merge-mining the tail size must be equal or greater than 169 bytes, but less than 233 bytes or the midstate is in fact the SHA256 initial state, while for a header that does have the merge mining tag, the tail size must be less than 169 bytes. This last check is the same as how the current check *ProofOfWorkRule* performs as: